### PR TITLE
feat: adds teams pagination, refactors tests

### PR DESCRIPTION
**Pull request by johnytiago**
This PR makes `cms/teams` obsolete for now.
Also, tests may fail while older versions of the codebase run simultaneous on the CI.
Once merged, no `teams` or `users` will be left on the CMS after testing.

### CODE REVIEW

File: apps/asap-server/src/controllers/teams.ts [hunk 1]
```diff
@@ -37,28 +37,68 @@ const transformUser = (users: CMSUser[], teamId: string): TeamMember[] =>
...
+      skip: (page - 1) * pageSize,
+      sort: [{ path: 'data.displayName.iv' }],
+    });
+
     const teamUsers = await Promise.all(
-      teams.map((team) => this.cms.users.fetchByTeam(team.id)),
+      teams.map((team) =>
+        this.users
+          .fetch({
+            take: undefined,
```
Comment (fampinheiro) [inline apps/asap-server/src/controllers/teams.ts:66]: Do we know what is the default value for this?

File: apps/asap-server/src/controllers/teams.ts [hunk 2]
```diff
@@ -37,28 +37,68 @@ const transformUser = (users: CMSUser[], teamId: string): TeamMember[] =>
...
+
+    return {
+      total,
+      items: teamItems,
+    };
   }
 
   async fetchById(teamId: string): Promise<TeamResponse> {
-    const [notFound, team] = await Intercept(this.cms.teams.fetchById(teamId));
+    const [notFound, team] = await Intercept(this.teams.fetchById(teamId));
```
Comment (fampinheiro) [inline apps/asap-server/src/controllers/teams.ts:88]: Not found is handled by the squidex wrapper.

```suggestion
    const team = await this.teams.fetchById(teamId);
```

File: apps/asap-server/src/controllers/teams.ts [hunk 3]
```diff
@@ -37,28 +37,68 @@ const transformUser = (users: CMSUser[], teamId: string): TeamMember[] =>
...
     if (notFound) {
       throw Boom.notFound();
     }
```
Comment (fampinheiro) [inline apps/asap-server/src/controllers/teams.ts:91]:
```suggestion
```

File: apps/asap-server/src/controllers/teams.ts (hunk 1 above)
Comment (johnytiago) [reply apps/asap-server/src/controllers/teams.ts:66]: It's 8, but in this case I don't want to limit it. I could potentially leave a big enough team size if we prefer

Comment (fampinheiro) [reply apps/asap-server/src/controllers/teams.ts:66]: Does this snippet does the same then?

```suggestion
```

**Comment by johnytiago** [comment]
You can consider that the tests are passing at this point. 
Ready for merge

**Review by fampinheiro — APPROVED** [review]

**Pull request merged**